### ISSUE TITLE
AUTO LOGIN  - - - - FASTER TOOLTIP HOVER

### ISSUE THREAD

**Issue by kstandif** [ace-lol/Ace#19]
Hello pls kindly consider an AUTO LOGIN  function 

as well as FASTER  TOOLTIPS when hovering (for example, .5 second delay over mastery hover)

THANK YOU!

**Comment by jaspervdm** [comment]
Auto login would have security issues. You would have to deal with storing the login details locally, in a secure way. Not saying its technically impossible but it may be undesirable.

**Comment by Everesty** [comment]
We had a bit of a discussion about auto-login in Discord. Here's the overview of why that's a bad idea:

*tl;dr at bottom.*

We'd have to store the user's login information on their computer somehow and that would mean it's somehow comprimisable by other people. We _could_ (and, of course, _would_ if the case arose) encrypt the data, but whatever key we use will be accessible by other third-parties given we have to decrypt without asking the user for the key. So we're back to the start with the issue of the information being stolen and accessible.

Modernly, this issue is partially helped by saving a token, rather than credentials. The server then remembers that the token is associated with something unique to the user (their IP address, perhaps) and who the user is.

However this isn't really an option for us because if we wanted to implement this then we'd have to have a server that stores user credentials and somehow sends the credentials or a session back to the user to use. This is infeasible for two reasons:

1. First and foremost, this is _suuuuuper_ sketchy. I wouldn't feel safe holding user information like this, and I doubt users would be comfortable knowing this information either. Morals.
2. We don't have a good way to check user credentials to be correct. We don't want to store incorrect user credentials because that can be abused to waste storage. So to authenticate the credentials, we'd need to ask Riot's servers. However, we'd eventually be limited to only a few checks per some amount of time.
  a. Using proxies isn't a very good solution because one user can technically just keep asking us to try new credentials that don't work and we end up with a bunch of proxies that Riot's servers will refuse login requests to.
  b. The service could also be exploited by people trying to bruteforce into accounts: whenever we finally can authenticate an account a token is sent back; to a bruteforcer, whenever a token is recieved, the credentials that were sent worked.

tl;dr: Auto login is nearly impossible to do securely and practically violates some morals in the process.

**Comment by Onurtag** [comment]
This would be dangerous for RIOT and could get the whole project blacklisted.

**Comment by FurkanS1821** [comment]
Take a look at #11 before opening this.
